internal/controllers/preflightvalidation_reconciler.go

```diff
@@ -100,14 +101,16 @@ func (r *preflightValidationReconciler) Reconcile(ctx context.Context, pv *v1bet
 		return ctrl.Result{}, fmt.Errorf("failed to get modules' data: %v", err)
 	}
 
-	err = r.helper.updateStatus(ctx, modsWithMapping, modsWithoutMapping, pv)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to update PreflightValidation's status: %v", err)
+	if micOpRes, err := r.helper.processPreflightValidation(ctx, modsWithMapping, pv); err != nil {
```

Comment:
Why do we need to switch places between updateStatus  and prcoessPreflightValidation?

Reply:
Because we need to validate the CreateOrPatch result before to understand if checking the status should be skipped or not.

Reply:
checking status knows how to handle missing MIC, so that is not supposed to be an issue. It also set the initial statuses of the existing modules in the preflight, so it makes sense for it ot be in first place